### PR TITLE
Added support for the flexible server deployment model in Azure.MySQL.GeoRedundantBackup

### LINKED ISSUE #1840: Azure Database for MySQL Flexible Servers should have geo-redundant backup configured
# Rule request

## Suggested rule change

As mentioned in #1836 

Lets add support in the rule `Azure.MySQL.GeoRedundantBackup` for flexible servers as well which is of the resource type `Microsoft.DBforMySQL/flexibleServers`.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.DBforMySQL/flexibleServers]**

## Additional context

[Backup and restore in Azure Database for MySQL Flexible Server](https://learn.microsoft.com/azure/mysql/flexible-server/concepts-backup-restore)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformysql/flexibleservers)

### PR DESCRIPTION
**Pull request by bengeset96**
## PR Summary

Fixes #1840 

Added support for the flexible server deployment model in `Azure.MySQL.GeoRedundantBackup` rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/en/rules/Azure.MySQL.GeoRedundantBackup.md [hunk 1]
```diff
@@ -94,12 +184,14 @@ resource mysqlDbServer 'Microsoft.DBforMySQL/servers@2017-12-01' = {
...
+For the Single Server deployment model, it runs only against `'General Purpose'` and `'Memory Optimized'` tiers. The `'Basic'` tier does not support geo-redundant backup storage.
 
 ## LINKS
 
 - [Target and non-functional requirements](https://learn.microsoft.com/azure/architecture/framework/resiliency/design-requirements)
-- [Backup and restore in Azure Database for MySQL](https://learn.microsoft.com/azure/mysql/single-server/concepts-backup)
-- [Azure template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformysql/servers)
+- [Backup and restore in Azure Database for MySQL flexible servers](https://learn.microsoft.com/azure/mysql/flexible-server/concepts-backup-restore)
+- [Backup and restore in Azure Database for MySQL single servers](https://learn.microsoft.com/azure/mysql/single-server/concepts-backup)
+- [Azure template reference flexible servers](https://learn.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/flexibleservers)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.MySQL.GeoRedundantBackup.md:196]:
```suggestion
- [Azure template reference flexible servers](https://learn.microsoft.com/azure/templates/microsoft.dbforpostgresql/flexibleservers)
```

File: docs/en/rules/Azure.MySQL.GeoRedundantBackup.md [hunk 2]
```diff
@@ -14,6 +14,24 @@ Azure Database for MySQL should store backups in a geo-redundant storage.
...
+
+#### Moving from other backup storage options to geo-redundant backup storage
+
+You can move your existing backups storage to geo-redundant storage using the following suggested ways:
+
+- **Moving from locally redundant to geo-redundant backup storage** - In order to move your backup storage from locally redundant storage to geo-redundant storage, you can change the Compute + Storage server configuration from Azure portal to enable Geo-redundancy for the locally redundant source server. Same Zone Redundant HA servers can also be restored as a geo-redundant server in a similar fashion as the underlying backup storage is locally redundant for the same.
+
+- **Moving from zone-redundant to geo-redundant backup storage** - Azure Database for MySQL does not support zone-redundant storage to geo-redundant storage conversion through Compute + Storage settings change or point-in-time restore operation. In order to move your backup storage from zone-redundant storage to geo-redundant storage, creating a new server and migrating the data using dump and restore is the only supported option.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.MySQL.GeoRedundantBackup.md:31]: Can we reduce this down? Ideally I don't want to duplicate the Azure docs, we just need to briefly explain the benefits so the customer understands why. They should refer back to the Azure docs if there is a detailed migration plan required.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 Nice work on the updates to the rule. I don't think we need too much extra detail from the original description as geo-redundant backups mostly work the same, just call out the slight differences as they relate to the flexible server.

File: docs/en/rules/Azure.MySQL.GeoRedundantBackup.md (hunk 2 above)
Comment (bengeset96) [reply docs/en/rules/Azure.MySQL.GeoRedundantBackup.md:31]: @BernieWhite 

Yes we can, I will adjust.

**Comment by bengeset96** [comment]
@BernieWhite 

I have updated the rule documentation. Could you check to see if we are aligned?

**Review by BernieWhite — APPROVED** [review]
Great work @bengeset96 ! Documentation updates are much clearer now. Thanks for your hard work.

**Pull request merged**